docarray/doc_index/backends/hnswlib_doc_index.py

```diff
@@ -0,0 +1,400 @@
+import hashlib
+import os
+import sqlite3
+from dataclasses import dataclass, field
+from pathlib import Path
+from typing import (
+    TYPE_CHECKING,
+    Any,
+    Dict,
+    Generic,
+    List,
+    Sequence,
+    Tuple,
+    Type,
+    TypeVar,
+    Union,
+    cast,
+)
+
+import hnswlib
+import numpy as np
+
+import docarray.typing
+from docarray import BaseDocument, DocumentArray
+from docarray.doc_index.abstract_doc_index import (
+    BaseDocumentIndex,
+    FindResultBatched,
+    _Column,
+)
+from docarray.proto import DocumentProto
+from docarray.utils.filter import filter as da_filter
+from docarray.utils.find import FindResult
+from docarray.utils.misc import is_np_int, torch_imported
+
+if TYPE_CHECKING:
+
+    def composable(fn):  # type: ignore
+        return fn
+
+else:
+    # static type checkers do not like callable objects as decorators
+    from docarray.doc_index.abstract_doc_index import composable
+
+TSchema = TypeVar('TSchema', bound=BaseDocument)
+T = TypeVar('T', bound='HnswDocumentIndex')
+
+HNSWLIB_PY_VEC_TYPES = [list, tuple, np.ndarray]
```

Comment:
what about basic python type like `str`, `int`...

Reply:
I still need to add support for those, but will do that in a separate PR.